engine/apps/alerts/models/alert_group.py

```diff
@@ -612,11 +612,37 @@ def _update_metrics(self, organization_id, previous_state, state):
         """Update metrics cache for response time and state as needed."""
         update_metrics_for_alert_group.apply_async((self.id, organization_id, previous_state, state))
 
-    def acknowledge_by_user(self, user: User, action_source: typing.Optional[ActionSource] = None) -> None:
+    def update_state_by_backsync(self, new_state: AlertGroupState) -> None:
+        if self.state == new_state:
+            return
+        logger.debug(f"Update state {self.state} -> {new_state} for alert_group {self.pk}")
+        if new_state == AlertGroupState.FIRING:
+            if self.state == AlertGroupState.ACKNOWLEDGED:
+                self.un_acknowledge_by_user_or_backsync(action_source=ActionSource.BACKSYNC)
+            elif self.state == AlertGroupState.RESOLVED:
+                self.un_resolve_by_user_or_backsync(action_source=ActionSource.BACKSYNC)
+            elif self.state == AlertGroupState.SILENCED:
+                self.un_silence_by_user_or_backsync(action_source=ActionSource.BACKSYNC)
+        elif new_state == AlertGroupState.ACKNOWLEDGED:
+            self.acknowledge_by_user_or_backsync(action_source=ActionSource.BACKSYNC)
+        elif new_state == AlertGroupState.RESOLVED:
+            self.resolve_by_user_or_backsync(action_source=ActionSource.BACKSYNC)
+        elif new_state == AlertGroupState.SILENCED:
+            self.silence_by_user_or_backsync(action_source=ActionSource.BACKSYNC)
```

Comment:
+1, I think it is clearer now what happen when we get an update (and it is nice to reuse the existing methods too)